src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -636,14 +636,28 @@ export default class DefaultDeviceController
       return;
     }
 
+    const stream = element.srcObject as MediaStream;
     // TODO: implement MediaDestroyer to provide single release MediaStream function
-    this.releaseMediaStream(element.srcObject as MediaStream);
-    DefaultVideoTile.disconnectVideoStreamFromVideoElement(element, false);
-    DefaultVideoTile.connectVideoStreamToVideoElement(
-      this.activeDevices['video'].stream,
-      element,
-      true
-    );
+    // The video stream should be released only if the video stream in the HTML element is different from current stream.
+    // When a filter is applied the existing stream is reused and in that case the stream should not be released.
+    if (stream && this.activeDevices['video'].stream !== stream) {
+      this.releaseMediaStream(element.srcObject as MediaStream);
+      DefaultVideoTile.disconnectVideoStreamFromVideoElement(element, false);
+    }
+
+    if (this.chosenVideoTransformDevice) {
+      DefaultVideoTile.connectVideoStreamToVideoElement(
+        this.chosenVideoTransformDevice.outputMediaStream,
```

Comment:
What happens if you choose a filter when there is no video preview stream? How does it get the inner stream?

Reply:
The code in that scenario will not call this method and it will get to the inner stream by calling `chooseInputIntrinsicDevice`. `chooseInputIntrinsicDevice` is called by `selectVideoFilterByName` and that is connected to UI element's event handler. This is at a high level and I skipped some calls in this explanation.